src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -558,6 +579,122 @@ def __init__(self, launch_string):
+        out = subprocess.check_output(
+            ["squeue", "-j", f"{self._job_id}", "-o", '"%T"', "-h"]
+        )
+        return out.decode().strip().strip('"')
+
+    def cancel(self, timeout: int = 60) -> bool:
+        if self.done():
+            return False
+        subprocess.run(["scancel", f"{self._job_id}"])
+        for _ in range(timeout):
+            if self.cancelled():
+                return True
+        return False
+
+    def cancelled(self) -> bool:
+        return self._get_state() == "CANCELLED"
+
+    def running(self) -> bool:
+        return self._get_state() == "RUNNING" and self._future.done()
+
+    def pending(self) -> bool:
+        return self._future.running()
+
+    def done(self) -> bool:
+        return self._get_state() in ["", "CANCELLED", "COMPLETED"]
+
+    def result(self, timeout=None):
+        return self._future.result(timeout)
+
+    def exception(self, timeout=None):
+        return self._future.exception(timeout)
+
+    def add_done_callback(self, fn):
+        self._future.add_done_callback(fn)
+
+
+class SlurmLauncher:
+    def __init__(
+        self,
+        argvals,
+        meshing_mode,
+        show_gui,
+        additional_arguments,
+        env,
+        topy,
+        journal_file_names,
+        new_session,
+        cleanup_on_exit,
+        start_transcript,
+    ):
+        self._argvals = argvals
+        self._meshing_mode = meshing_mode
+        self._show_gui = show_gui
+        self._additional_arguments = additional_arguments
+        self._env = env
+        self._topy = topy
+        self._journal_file_names = journal_file_names
+        self._new_session = new_session
+        self._cleanup_on_exit = cleanup_on_exit
+        self._start_transcript = start_transcript
+
+    def _prepare(self):
+        self._server_info_file_name = _get_server_info_file_name(use_tmpdir=False)
+        self._argvals.update(self._argvals["scheduler_options"])
+        launch_cmd = _generate_launch_string(
+            self._argvals,
+            self._meshing_mode,
+            self._show_gui,
+            self._additional_arguments,
+            self._server_info_file_name,
+        )
+
+        self._sifile_last_mtime = Path(self._server_info_file_name).stat().st_mtime
+        if self._env is None:
+            self._env = {}
+        kwargs = _get_subprocess_kwargs_for_fluent(self._env, self._argvals)
+        launch_cmd += _build_journal_argument(self._topy, self._journal_file_names)
+
+        logger.debug(f"Launching Fluent with command: {launch_cmd}")
+        proc = subprocess.Popen(launch_cmd, **kwargs)
+        slurm_job_id = _get_slurm_job_id(proc)
+        logger.info(f"Slurm job id = {slurm_job_id}")
+        self._argvals["slurm_job_id"] = slurm_job_id
+        return slurm_job_id
+
+    def _launch(self, slurm_job_id) -> Union[Meshing, PureMeshing, Solver, SolverIcing]:
+        _await_fluent_launch(self._server_info_file_name, -1, self._sifile_last_mtime)
```

Comment:
The way that `_await_fluent_launch` is being called here, there is no timeout and the logging calls aren't ideal:
![image](https://github.com/ansys/pyfluent/assets/132297401/5ce26ca5-0794-4796-8453-7191853efe5a)

Should we support the existing default timeout (currently 1 minute), with user-configurable Slurm queue timeout through the existing `launch_fluent(start_timeout=...)` argument? Could change the behavior so that setting this timeout to 0 means it will wait indefinitely (with adjusted logging reports without the "X seconds remaining" part), or have a different default behavior when launching Fluent in a Slurm queue (but still respecting what the user specifies in `start_timeout`).

Reply:
For Slurm, we need to set the default for `start_timeout` such that it will wait indefinitely. I don't know what is a good value for a default, I've heard things can wait for a week.

Reply:
I'm not sure what is a good value for the default, so I'll set the default of `start_timeout` such that it will wait indefinitely.